### PR TITLE
Minor fix: Add steps as Optional in __init__

### PR DESCRIPTION
**Pull request by geetikakay**
##### Short description:
Fix for adding optional to steps.

## Summary by CodeRabbit

- **New Features**
	- Enhanced flexibility in the `Task` class by allowing the omission of the `steps` parameter during instantiation.
- **Bug Fixes**
	- Improved handling of cases where no steps are provided, ensuring smoother functionality.

### CODE REVIEW

**Comment by redhat-qe-bot1** [comment]
Report bugs in [Issues](https://github.com/myakove/github-webhook-server/issues)

The following are automatically added:
 * Add reviewers from OWNER file (in the root of the repository) under reviewers section.
 * Set PR size label.
 * New issue is created for the PR. (Closed when PR is merged/closed)
 * Run [pre-commit](https://pre-commit.ci/) if `.pre-commit-config.yaml` exists in the repo.

Available user actions:
 * To mark PR as WIP comment `/wip` to the PR, To remove it from the PR comment `/wip cancel` to the PR.
 * To block merging of PR comment `/hold`, To un-block merging of PR comment `/hold cancel`.
 * To mark PR as verified comment `/verified` to the PR, to un-verify comment `/verified cancel` to the PR.
        verified label removed on each new commit push.
 * To cherry pick a merged PR comment `/cherry-pick <target branch to cherry-pick to>` in the PR.
    * Multiple target branches can be cherry-picked, separated by spaces. (`/cherry-pick branch1 branch2`)
    * Cherry-pick will be started when PR is merged
 * To build and push container image command `/build-and-push-container` in the PR (tag will be the PR number).
    * You can add extra args to the Podman build command
        * Example: `/build-and-push-container --build-arg OPENSHIFT_PYTHON_WRAPPER_COMMIT=<commit_hash>`
 * To add a label by comment use `/<label name>`, to remove, use `/<label name> cancel`
 * To assign reviewers based on OWNERS file use `/assign-reviewers`
 * To check if PR can be merged use `/check-can-merge`

<details>
<summary>Supported /retest check runs</summary>

 * `/retest tox`: Retest tox
 * `/retest python-module-install`: Retest python-module-install

</details>

<details>
<summary>Supported labels</summary>

 * hold
 * verified
 * wip
 * lgtm

</details>

File: ocp_resources/task.py [hunk 1]
```diff
@@ -12,7 +12,7 @@ class Task(NamespacedResource):
 
     def __init__(
         self,
-        steps: List[Dict[str, Any]],
+        steps: Optional[List[Dict[str, Any]]] = None,
```
Comment (myakove) [inline ocp_resources/task.py:15]: From the docs `steps` is Required

```
Required:
[apiVersion](https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#required-fields) - Specifies the API version. For example, tekton.dev/v1beta1.
[kind](https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#required-fields) - Identifies this resource object as a Task object.
[metadata](https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#required-fields) - Specifies metadata that uniquely identifies the Task resource object. For example, a name.
[spec](https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#required-fields) - Specifies the configuration information for this Task resource object.
[steps](https://tekton.dev/docs/pipelines/tasks/#defining-steps) - Specifies one or more container images to run in the Task.
```

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/task.py (hunk 1 above)
Comment (myakove) [reply ocp_resources/task.py:15]: nm, we enforce it inside `to_dict` :)

**Review by myakove — APPROVED** [review]

**Pull request merged**